cli/src/api/mod.rs

```diff
@@ -110,7 +123,14 @@ impl PhylumApi {
         let body = response.text().await?;
 
         if !status_code.is_success() {
-            let err = ResponseError { body, code: status_code };
+            let details = if let Ok(err) = serde_json::from_str::<ApiJsonError>(&body) {
+                log::debug!("Error ID: {}", err.error.error_id);
+                err.error.description
+            } else {
+                body
+            };
```

Comment:
I'm debating if this is more readable as `.map(...).unwrap_or(...)` 🤷‍♂️

```
            let details = serde_json::from_str::<ApiJsonError>(&body)
                .map(|err| {
                    log::debug!("Error ID: {}", err.error.error_id);
                    err.error.description
                })
                .unwrap_or(body);
```

Reply:
I don't think so. I'd usually go for a match when having two branches anyway, but in this scenario it doesn't really matter and your solution is technically shorter I think.